skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/aibot/ChatWindowUi.kt

```diff
@@ -0,0 +1,178 @@
+      LazyColumn(modifier = Modifier.weight(1f), reverseLayout = true) {
+        items(state.messages.reversed()) { message ->
+          Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = if (message.isMe) Arrangement.End else Arrangement.Start,
+          ) {
+            ChatBubble(message)
+          }
+        }
+      }
+      ConversationField(
+        modifier = Modifier,
+        onSendMessage = { userMessage ->
+          state.eventSink(ChatScreen.Event.SendMessage(userMessage))
+        },
+      )
+    }
+  }
+
+  @Composable
+  fun ConversationField(modifier: Modifier = Modifier, onSendMessage: (String) -> Unit) {
+    var textValue by remember { mutableStateOf(TextFieldValue()) }
+    val isTextNotEmpty = textValue.text.isNotBlank()
+
+    fun sendMessage() {
+      if (isTextNotEmpty) {
+        onSendMessage(textValue.text)
+        textValue = TextFieldValue("")
+      }
+    }
+
+    Row(
+      modifier.padding(4.dp).height(IntrinsicSize.Min),
+      horizontalArrangement = Arrangement.Center,
+      verticalAlignment = Alignment.Bottom,
+    ) {
+      // user text input. enter sends the message and clears text
+      // shift + enter adds a new line and expands the text field
+      TextArea(
+        value = textValue,
+        onValueChange = { newText -> textValue = newText },
+        modifier =
+          Modifier.weight(1f).heightIn(min = 56.dp).padding(4.dp).onPreviewKeyEvent { event ->
+            when {
+              (event.key == Key.Enter || event.key == Key.NumPadEnter) &&
+                event.type == KeyEventType.KeyDown -> {
+                if (event.isShiftPressed) {
+                  val newText =
+                    textValue.text.replaceRange(
+                      textValue.selection.start,
+                      textValue.selection.end,
+                      "\n",
+                    )
+                  val newSelection = TextRange(textValue.selection.start + 1)
+                  textValue = TextFieldValue(newText, newSelection)
+                  true
+                } else {
+                  sendMessage()
+                  true
+                }
+              }
+              else -> false
+            }
+          },
+        placeholder = { Text("Start your conversation") },
+        keyboardActions = KeyboardActions.Default,
+        maxLines = Int.MAX_VALUE,
+      )
+      Column(Modifier.fillMaxHeight(), verticalArrangement = Arrangement.Center) {
+        // button will be disabled if there is no text
+        IconButton(
+          modifier = Modifier.padding(4.dp).enabled(isTextNotEmpty),
+          onClick = {
+            if (isTextNotEmpty) {
+              sendMessage()
+            }
+          },
+          enabled = isTextNotEmpty,
+        ) {
+          Icon(
+            painter = painterResource("/drawable/send.svg"),
+            contentDescription = "Send",
+            modifier = Modifier.size(20.dp),
```

Comment:
For this I would recommend using `Modifier.minimumInteractiveComponentSize()`, which is more dynamic and will respect user accessibility settings too

Reply:
Good point. Maybe look at their docs and see if they have anything comparable?

Reply:
Could I use `wrapContentSize()` and set `unbounded = true` to account for accessibility? 

According to the docs: 
"Allow the content to measure at its desired size without regard for the incoming measurement minimum width or minimum height constraints, and, if unbounded is true, also without regard for the incoming maximum constraints. If the content's measured size is smaller than the minimum size constraint, align it within that minimum sized space. If the content's measured size is larger than the maximum size constraint (only possible when unbounded is true), align within the maximum space."